cmd/clusterctl/pkg/client/cluster/upgrader_test.go

```diff
@@ -428,6 +428,55 @@ func Test_providerUpgrader_Plan(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "Single Management group, no multi-tenancy, upgrade for two contracts, but the upgrade for the second one is partially available",
+			fields: fields{
+				// config for two providers
+				reader: test.NewFakeReader().
+					WithProvider("core", clusterctlv1.CoreProviderType, "https://somewhere.com").
+					WithProvider("infra", clusterctlv1.InfrastructureProviderType, "https://somewhere.com"),
+				// two provider repositories, the first with a new version for current v1alpha3 contract and a new version for the v1alpha4 contract, the second without new releases
+				repository: map[string]repository.Repository{
+					"core": test.NewFakeRepository().
+						WithVersions("v1.0.0", "v1.0.1", "v2.0.0").
+						WithMetadata("v2.0.0", &clusterctlv1.Metadata{
+							ReleaseSeries: []clusterctlv1.ReleaseSeries{
+								{Major: 1, Minor: 0, Contract: "v1alpha3"},
+								{Major: 2, Minor: 0, Contract: "v1alpha4"},
+							},
+						}),
+					"infra": test.NewFakeRepository().
+						WithVersions("v2.0.0"). // no v1alpha3 or v1alpha3 new releases yet available for the infra provider (only the current release exists)
```

Comment:
Just wanted to clarify this comment. Is `infra v2.0.0` **not a v1a3 release**? Because in the metadata below we specify the ReleaseSeries for v2.0 to abide by the v1a3 contract.

Is there a documentation describing the concepts of contracts?

Reply:
```suggestion
						WithVersions("v2.0.0"). // no v1alpha3 or v1alpha3 new releases not yet available for the infra provider (only the current release exists)
```

Reply:
I believe I understand the concept in this PR. The core provider had a release for v1a4, however since the infra provider didn't have a release that abided to the v1a4 contract, the v1a4 upgrade plan for the core provider was dropped since "_all the provider in a management group are required to change contract at the same time_"

Ok this makes sense. 😄